pkg/skaffold/deploy/kpt.go

```diff
@@ -47,3 +56,53 @@ func (k *KptDeployer) Cleanup(ctx context.Context, out io.Writer) error {
 func (k *KptDeployer) Render(ctx context.Context, out io.Writer, builds []build.Artifact, offline bool, filepath string) error {
 	return nil
 }
+
+// getApplyDir returns the path to applyDir if specified by the user. Otherwise, getApplyDir
+// creates a hidden directory in place of applyDir.
+func (k *KptDeployer) getApplyDir(ctx context.Context) (string, error) {
+	if k.ApplyDir != "" {
+		if _, err := os.Stat(k.ApplyDir); os.IsNotExist(err) {
+			return "", err
+		}
+		return k.ApplyDir, nil
+	}
+
+	applyDir := ".kpt-hydrated"
```

Comment:
We would want to check the existence of .kpt-hydrated befor creation.

Reply:
If .kpt-hydrated already exists, os.MkdirAll does nothing and returns nil.